File: Backend/src/main/java/com/example/ecommerce/dto/BankTransactionDto.java
package com.example.ecommerce.dto;

import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.Setter;
import java.math.BigDecimal;

@Getter @Setter
@JsonIgnoreProperties(ignoreUnknown = true)
public class BankTransactionDto {
    private String gateway;
    
    @JsonProperty("transactiondate")
    @JsonAlias({"transactionDate", "TransactionDate"})
    private String transactiondate;
    
    @JsonProperty("accountnumber")
    private String accountnumber;
    
    @JsonProperty("subaccount")
    private String subaccount;
    
    private String code;
    private String content;
    
    @JsonProperty("transfertype")
    private String transfertype;
    
    @JsonProperty("transferamount")
    private BigDecimal transferamount;
    
    private BigDecimal accumulated;
    
    @JsonProperty("referencecode")
    private String referencecode;
    
    private String description;
}

File: New folder/Backend/Models/DTOs/BankTransactionDto.cs
namespace PTVBTPM.Models.DTOs
{
    public class BankTransactionDto
    {
        public string Gateway { get; set; } = string.Empty;
        public string Transactiondate { get; set; } = string.Empty;
        public string Accountnumber { get; set; } = string.Empty;
        public string? Subaccount { get; set; }
        public string? Code { get; set; }
        public string Content { get; set; } = string.Empty;
        public string Transfertype { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public decimal Transferamount { get; set; }
        public string Referencecode { get; set; } = string.Empty;
        public decimal Accumulated { get; set; }
    }
}

